src/main/java/org/openrewrite/java/testing/assertj/SimplifyChainedAssertJAssertions.java

```diff
@@ -0,0 +1,139 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.assertj;
+
+import lombok.AllArgsConstructor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+@AllArgsConstructor
+public class SimplifyChainedAssertJAssertions extends Recipe {
+    @Option(displayName = "AssertJ Assertion",
+            description = "The chained AssertJ assertion to move to dedicated assertion.",
+            example = "equals",
+            required = false)
+    @Nullable
+    String chainedAssertion;
+
+    @Option(displayName = "AssertJ Assertion",
+            description = "The AssertJ assert that should be replaced",
+            example = "isTrue",
+            required = false)
+    @Nullable
+    String assertToReplace;
+
+    @Option(displayName = "AssertJ Assertion",
+            description = "The AssertJ method to migrate to.",
+            example = "isEqualTo",
+            required = false)
+    @Nullable
+    String dedicatedAssertion;
+
+    @Override
+    public String getDisplayName() {
+        return "Simplify AssertJ chained assertions";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-5838");
+    }
+
+    @Override
+    public String getDescription() {
+        return "Many AssertJ chained assertions have dedicated assertions that function the same. " +
+                "It is best to use the dedicated assertions.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new SimplifyChainedAssertJAssertionsVisitor();
+    }
+
+    private class SimplifyChainedAssertJAssertionsVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private final MethodMatcher ASSERT_THAT_MATCHER = new MethodMatcher("org.junit.jupiter.api.Assertions assertThat(..)");
+        private final MethodMatcher CHAINED_ASSERT_MATCHER = new MethodMatcher("java..* " + chainedAssertion + "(..)");
+        private final MethodMatcher ASSERT_TO_REPLACE = new MethodMatcher("org.assertj.core.api.*" + assertToReplace + "(..)");
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation mi, ExecutionContext ctx) {
+            //  assert has assertion
+            if (!ASSERT_TO_REPLACE.matches(mi) || mi.getSelect() == null) {
```

Comment:
What breaks if we remove the first part of this conditional? I'd expect nothing.
```suggestion
            if (mi.getSelect() == null) {
```